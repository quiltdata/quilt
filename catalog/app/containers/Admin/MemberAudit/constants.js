import { createActions } from 'utils/reduxTools';

export const REDUX_KEY = 'app/Admin/MemberAudit';

export const actions = createActions(REDUX_KEY,
  'GET',
  'GET_RESPONSE',
); // eslint-disable-line function-paren-newline
