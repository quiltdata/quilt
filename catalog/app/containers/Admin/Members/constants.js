import { createActions } from 'utils/reduxTools';

export const REDUX_KEY = 'app/Admin/Members';

export const actions = createActions(REDUX_KEY,
  'ADD',
  'ADDED',
  'GET',
  'GET_RESPONSE',
  'DISABLE',
  'DISABLE_RESPONSE',
  'ENABLE',
  'ENABLE_RESPONSE',
  'RESET_PASSWORD',
  'RESET_PASSWORD_RESPONSE',
); // eslint-disable-line function-paren-newline
