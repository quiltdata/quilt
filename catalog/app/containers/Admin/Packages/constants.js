import { createActions } from 'utils/reduxTools';

export const REDUX_KEY = 'app/Admin/Packages';

export const actions = createActions(REDUX_KEY,
  'GET',
  'GET_RESPONSE',
); // eslint-disable-line function-paren-newline
