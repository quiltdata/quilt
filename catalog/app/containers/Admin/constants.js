/*
 *
 * Admin constants
 *
 */

export const ADD_MEMBER = 'app/Admin/ADD_MEMBER';
export const MEMBER_ADDED = 'app/Admin/MEMBER_ADDED';
export const GET_MEMBERS = 'app/Admin/GET_MEMBERS';
export const GET_MEMBERS_RESPONSE = 'app/Admin/GET_MEMBERS_RESPONSE';
export const GET_MEMBER_AUDIT = 'app/Admin/GET_MEMBER_AUDIT';
export const GET_MEMBER_AUDIT_RESPONSE = 'app/Admin/GET_MEMBER_AUDIT_RESPONSE';
export const DISABLE_MEMBER = 'app/Admin/DISABLE_MEMBER';
export const DISABLE_MEMBER_RESPONSE = 'app/Admin/DISABLE_MEMBER_RESPONSE';
export const ENABLE_MEMBER = 'app/Admin/ENABLE_MEMBER';
export const ENABLE_MEMBER_RESPONSE = 'app/Admin/ENABLE_MEMBER_RESPONSE';
export const RESET_MEMBER_PASSWORD = 'app/Admin/RESET_MEMBER_PASSWORD';
export const RESET_MEMBER_PASSWORD_RESPONSE = 'app/Admin/RESET_MEMBER_PASSWORD_RESPONSE';
export const GET_PACKAGES = 'app/Admin/GET_PACKAGES';
export const GET_PACKAGES_RESPONSE = 'app/Admin/GET_PACKAGES_RESPONSE';
export const GET_PACKAGE_AUDIT = 'app/Admin/GET_PACKAGE_AUDIT';
export const GET_PACKAGE_AUDIT_RESPONSE = 'app/Admin/GET_PACKAGE_AUDIT_RESPONSE';
