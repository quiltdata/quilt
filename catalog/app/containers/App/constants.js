/* App constants */
import { fromJS } from 'immutable';


// user object prevents authBarSelector from failing on startup
export const initialState = fromJS({});

export const DEFAULT_LOCALE = 'en';

// TODO: move to config
export const intercomAppId = 'eprutqnr';

export const REDUX_KEY = 'app/App';

export const GET_LOG = 'app/Profile/GET_LOG';
export const GET_LOG_ERROR = 'app/Profile/GET_LOG_ERROR';
export const GET_LOG_SUCCESS = 'app/Profile/GET_LOG_SUCCESS';

export const GET_PACKAGE = 'app/App/GET_PACKAGE';
export const GET_PACKAGE_ERROR = 'app/App/GET_PACKAGE_ERROR';
export const GET_PACKAGE_SUCCESS = 'app/App/GET_PACKAGE_SUCCESS';

export const GET_MANIFEST = 'app/App/GET_MANIFEST';
export const GET_MANIFEST_ERROR = 'app/App/GET_MANIFEST_ERROR';
export const GET_MANIFEST_SUCCESS = 'app/App/GET_MANIFEST_SUCCESS';

export const GET_TRAFFIC = 'app/App/GET_TRAFFIC';
export const GET_TRAFFIC_RESPONSE = 'app/App/GET_TRAFFIC_RESPONSE';

export const SET_SEARCH_TEXT = 'app/App/SET_SEARCH_TEXT';

export const START = 'app/App/START';
