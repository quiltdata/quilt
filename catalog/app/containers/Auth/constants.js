import { createActions } from 'utils/reduxTools';

export const REDUX_KEY = 'app/Auth';

export const states = [
  'SIGNED_OUT',
  'SIGNING_IN',
  'SIGNED_IN',
  'REFRESHING',
];

export const waitingStates = [
  'SIGNING_IN',
  'REFRESHING',
];

export const actions = createActions(REDUX_KEY,
  'SIGN_UP',
  'SIGN_IN',
  'SIGN_IN_RESULT',
  'SIGN_OUT',
  'SIGN_OUT_RESULT',
  'CHECK',
  'REFRESH',
  'REFRESH_SUCCESS',
  'REFRESH_ERROR',
  'AUTH_LOST',
); // eslint-disable-line function-paren-newline

// DEBUG - 10 second expiry
// export const LATENCY_SECONDS = 35990;
export const LATENCY_SECONDS = 20;

export const ERROR_NOTIFICATION_TTL = 20000;
export const ERROR_REDIRECT_PATH = '/';
