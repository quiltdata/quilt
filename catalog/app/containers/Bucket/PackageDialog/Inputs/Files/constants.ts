import cfg from 'constants/config'

export const MAX_UPLOAD_SIZE = 20 * 1000 * 1000 * 1000 // 20GB
// XXX: keep in sync w/ the backend
// NOTE: these limits are lower than the actual "hard" limits on the backend
export const MAX_S3_SIZE = cfg.chunkedChecksums
  ? 5 * 10 ** 12 // 5 TB
  : 50 * 10 ** 9 // 50 GB
export const MAX_FILE_COUNT = 1000
