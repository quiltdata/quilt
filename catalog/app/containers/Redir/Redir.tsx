import invariant from 'invariant'
import * as React from 'react'
import { useParams } from 'react-router-dom'
import * as M from '@material-ui/core'

import Empty from 'components/Empty'
import Layout from 'components/Layout'
import * as UriResolver from 'containers/UriResolver'
import MetaTitle from 'utils/MetaTitle'
import * as PackageUri from 'utils/PackageUri'
import { BaseError } from 'utils/error'

interface OpenInDesktopProps {
  href: string
}

function OpenInDesktop({ href }: OpenInDesktopProps) {
  return (
    <M.Button color="primary" component="a" href={href} variant="contained">
      Open QuiltSync
    </M.Button>
  )
}

const useStyles = M.makeStyles((t) => ({
  root: {
    margin: t.spacing(2, 0),
  },
}))

const REDIRECT_TIMEOUT = 1000

export default function Redir() {
  const params = useParams<{ uri?: string }>()

  invariant(params.uri, '`uri` must be defined')

  const classes = useStyles()

  const decoded = decodeURIComponent(params.uri)
  const uri = React.useMemo(() => UriResolver.parsePackageUriSafe(decoded), [decoded])

  const [redirecting, setRedirecting] = React.useState<PackageUri.PackageUri | null>(null)
  React.useEffect(() => {
    if (uri instanceof BaseError) return

    window.location.assign(decoded)

    const timeoutId = setTimeout(() => setRedirecting(uri), REDIRECT_TIMEOUT)
    return () => clearTimeout(timeoutId)
  }, [decoded, uri])

  if (redirecting) return <UriResolver.Redirect parsed={redirecting} decoded={decoded} />

  return (
    <Layout>
      <MetaTitle>Resolve a Quilt+ URI</MetaTitle>
      {uri instanceof BaseError ? (
        <Empty
          className={classes.root}
          primary={<OpenInDesktop href="quilt+s3://" />}
          secondary="Click the button to open QuiltSync"
          title="Failed to redirect"
        >
          <M.Typography color="error">
            Error parsing URI: {uri.msg || `${uri}`}
          </M.Typography>
        </Empty>
      ) : (
        <Empty
          className={classes.root}
          primary={<OpenInDesktop href={PackageUri.stringify(uri)} />}
          secondary="Click the button if you’re not redirected automatically."
          title="Redirecting…"
        />
      )}
    </Layout>
  )
}
