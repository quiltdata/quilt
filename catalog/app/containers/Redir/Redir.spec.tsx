import * as React from 'react'
import { ErrorBoundary, type FallbackProps } from 'react-error-boundary'
import { render, act } from '@testing-library/react'

import { bucketPackageTree } from 'constants/routes'
import * as NamedRoutes from 'utils/NamedRoutes'

import Redir from './Redir'

const FallbackComponent = ({ error }: FallbackProps) => (
  <span>Error: {error.message}</span>
)

jest.mock(
  'constants/config',
  jest.fn(() => ({})),
)

jest.mock(
  'components/Layout',
  jest.fn(() => ({ children }: React.PropsWithChildren<{}>) => (
    <div role="main">{children}</div>
  )),
)

const useParams = jest.fn(() => ({ uri: '' }) as Record<string, string>)

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useParams: jest.fn(() => useParams()),
  Redirect: jest.fn(({ to }: { to: string }) => `Redirect to ${to}`),
}))

jest.mock(
  '@material-ui/core',
  jest.fn(() => ({
    ...jest.requireActual('@material-ui/core'),
    Button: jest.fn(({ children, href }: React.PropsWithChildren<{ href: string }>) => (
      <a href={href}>{children}</a>
    )),
  })),
)

describe('containers/Redir/Redir', () => {
  beforeEach(() => {
    jest.spyOn(console, 'error').mockImplementation(jest.fn())
  })

  it('must have uri', () => {
    const { container } = render(
      <ErrorBoundary FallbackComponent={FallbackComponent}>
        <Redir />
      </ErrorBoundary>,
    )
    expect(container.firstChild).toMatchSnapshot()
  })

  it('shows waiting screen', () => {
    useParams.mockImplementationOnce(() => ({
      uri: 'quilt+s3://bucket#package=pkg/name@hash',
    }))
    const { container } = render(
      <NamedRoutes.Provider routes={{ bucketPackageTree }}>
        <Redir />
      </NamedRoutes.Provider>,
    )
    expect(container.firstChild).toMatchSnapshot()
  })

  it('shows error', () => {
    useParams.mockImplementationOnce(() => ({
      uri: 'invalid',
    }))
    const { container } = render(<Redir />)
    expect(container.firstChild).toMatchSnapshot()
  })

  it('redirects to package page', async () => {
    // TODO: spy on window.location.assign
    jest.useFakeTimers()
    useParams.mockImplementation(() => ({
      uri: 'quilt+s3://bucket#package=pkg/name@hash',
    }))
    const { container } = render(
      <NamedRoutes.Provider routes={{ bucketPackageTree }}>
        <Redir />
      </NamedRoutes.Provider>,
    )
    await act(() => jest.runAllTimersAsync())
    expect(container.firstChild).toMatchSnapshot()
    jest.useRealTimers()
  })
})
