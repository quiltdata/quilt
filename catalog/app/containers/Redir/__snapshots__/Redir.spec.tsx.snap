// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`containers/Redir/Redir must have uri 1`] = `
<span>
  Error: 
  \`uri\` must be defined
</span>
`;

exports[`containers/Redir/Redir redirects to package page 1`] = `"Redirect to /b/bucket/packages/pkg/name/tree/hash/?resolvedFrom=quilt%2Bs3%3A%2F%2Fbucket%23package%3Dpkg%2Fname%40hash"`;

exports[`containers/Redir/Redir shows error 1`] = `
<div
  role="main"
>
  <div
    className="makeStyles-root-2 makeStyles-root-1"
  >
    <h4
      className="MuiTypography-root MuiTypography-h4"
    >
      Failed to redirect
    </h4>
    <div
      className="makeStyles-actions-3"
    >
      <div
        className="makeStyles-primary-4"
      >
        <a
          href="quilt+s3://"
        >
          Open QuiltSync
        </a>
      </div>
      <p
        className="MuiTypography-root makeStyles-secondary-5 MuiTypography-body1"
      >
        Click the button to open QuiltSync
      </p>
    </div>
    <div
      className="makeStyles-main-6"
    >
      <p
        className="MuiTypography-root MuiTypography-body1 MuiTypography-colorError"
      >
        Error parsing URI: 
        unsupported protocol "null". "quilt+s3:" is currently the only supported protocol.
      </p>
    </div>
  </div>
</div>
`;

exports[`containers/Redir/Redir shows waiting screen 1`] = `
<div
  role="main"
>
  <div
    className="makeStyles-root-2 makeStyles-root-1"
  >
    <h4
      className="MuiTypography-root MuiTypography-h4"
    >
      Redirecting…
    </h4>
    <div
      className="makeStyles-actions-3"
    >
      <div
        className="makeStyles-primary-4"
      >
        <a
          href="quilt+s3://bucket#package=pkg/name@hash"
        >
          Open QuiltSync
        </a>
      </div>
      <p
        className="MuiTypography-root makeStyles-secondary-5 MuiTypography-body1"
      >
        Click the button if you’re not redirected automatically.
      </p>
    </div>
  </div>
</div>
`;
